### ISSUE TITLE
The Tag Value provided is invalid <tag value>

### ISSUE THREAD

**Issue by seankovacs** [aws-amplify/amplify-cli#6628]
**Describe the bug**
With my old Amp project, I made some changes to functions and tried to push updates via amplify push, using the latest CLI, and started receiving errors coming from something that wasn't even modified - DynamoDB tables. It came in the form of this error during push: "UPDATE_FAILED DynamoDBTable AWS::DynamoDB::Table Mon Feb 15 2021 11:01:43 GMT-0500 (Eastern Standard Time) The Tag Value provided is invalid Test's (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: Q5F501CTPI0M6POU9C3Q7PJVTFVV4KQNSO5AEMVJF66Q9ASUAAJG; Proxy: null)"

I checked the cloud formation templates and discovered that there weren't any tags in there, so I figured something else must be going on. Apparently a new tags.json is being created for new Amplify projects and in the absence of this file, it rolls its own defaults for old projects, which if you have invalid characters in your project-name like I do, you will get the error I'm getting above.

I added the tags.json file and changed the value of "user:Application" from "{project-name}" to something that wouldn't trip over itself. 

**Amplify CLI Version**
4.43.0

**To Reproduce**
1. Create a new Amplify app/backend from the web, using a single quote in the name. The web allows it, but the CLI does not!
2. Pull down the backend of that new app via CLI.
3. Add something new like NoSQL storage.
4. Push changes & you'll get the error.

**Expected behavior**
I expected an error free deployment. I expected the Cloudformation templates to have these invalid Tag values, but they didn't. I expected a lot less voodoo going on.

**Screenshots**
N/A

**Desktop (please complete the following information):**

- OS: Mac OSX 10.15.7
- Node Version. v14.15.0

**Log output**
```
CREATE_FAILED amplify-tests-staging-161557-storagedynamodd02928d-1FLWVGFOJH7EL AWS::CloudFormation::Stack Mon Feb 15 2021 11:24:11 GMT-0500 (Eastern Standard Time) The following resource(s) failed to create: [DynamoDBTable].                                                                                                                                          
CREATE_FAILED DynamoDBTable                                                    AWS::DynamoDB::Table       Mon Feb 15 2021 11:24:10 GMT-0500 (Eastern Standard Time) The Tag Value provided is invalid Test's (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: 6C9DH8L131BPTRIA1SAFJNENTVVV4KQNSO5AEMVJF66Q9ASUAAJG; Proxy: null)
⠦ Updating resources in the cloud. This may take a few minutes...

CREATE_FAILED storagedynamodd02928d AWS::CloudFormation::Stack Mon Feb 15 2021 11:24:18 GMT-0500 (Eastern Standard Time) Embedded stack arn:aws:cloudformation:us-east-1:841987401434:stack/amplify-tests-staging-161557-storagedynamodd02928d-1FLWVGFOJH7EL/24b5f4c0-6faa-11eb-9d87-0a54954134e7 was not successfully created: The following resource(s) failed to create: [DynamoDBTable]. 
⠙ Updating resources in the cloud. This may take a few minutes...
```

**Additional context**
Since a lot of projects most likely have characters that Tag values can't support, it might be best to strip non-alphanumeric characters from {project-name}.

**Comment by yuth** [comment]
@seankovacs Thank you for the report. 
> Create a new Amplify app/backend from the web, using a single quote in the name. The web allows it, but the CLI does not!

When you say create from the web do you mean by using the Admin UI (https://sandbox.amplifyapp.com/)?

We added tag validation in https://github.com/aws-amplify/amplify-cli/commit/31eb8ebff2fcbd215975c8ac05287d023d544c42 and its waiting for the release and should be included in next release (after 4.43.0). This should stop the deployment before a cloud-formation deployment is started.

**Comment by seankovacs** [comment]
> @seankovacs Thank you for the report.
> 
> > Create a new Amplify app/backend from the web, using a single quote in the name. The web allows it, but the CLI does not!
> 
> When you say create from the web do you mean by using the Admin UI (https://sandbox.amplifyapp.com/)?
> 
> We added tag validation in [31eb8eb](https://github.com/aws-amplify/amplify-cli/commit/31eb8ebff2fcbd215975c8ac05287d023d544c42) and its waiting for the release and should be included in next release (after 4.43.0). This should stop the deployment before a cloud-formation deployment is started.

Good question. I created it via the [AWS Amplify Console](https://console.aws.amazon.com/amplify/home?region=us-east-1#/). 
That's good to hear regarding the new update.

**Comment by yuth** [comment]
The fix has been release in Amplify CLI v4.40
